Code/Diversity_Histogram.py

```diff
@@ -65,15 +70,19 @@
                                                         Gene_Positive_Negative_Number['Total_Diauxic']/38)\
                                                         /(Gene_Positive_Negative_Number['Total_Diauxic']/38+\
                                                          Gene_Positive_Negative_Number['Total_Polarity']/123)
-
+# questions: 
+    # - where do 123 and 38 come from? 
```

Comment:
123 = Number of genes in the polarity module
38 = Number of genes in the diauxic shift module

Reply:
- I think that these numbers should be given by certain datafile you filtered out in your search. Like for example a variable that says `total_number_of_genes_polarity_module=len(datafile_Polarity_genes)=123` with an short explanation of how you got these numbers. 
- Are they given in your Interaction_Matrix dataset? like the unique number of genes correspoding to polarity are 123 and 38 for the diauxic shift module? If so then you can put them as : `total_number_of_genes_polarity_module=len(np.unique(Interaction_Matrix.loc[:,'Interactor_Module']))`

Reply:
The datafiles that could be used to get these numbers would ideally be '**Genes_Diauxic_Shift_Description.xlsx**' and '**Polarity_Related_Genes_Description.xlsx**', which contain the genes related to each module. I guess it could also be extracted from the Interaction_Matrix dataframe, but I think that would be more complicated/less interpretable since this information is more embedded in that dataset (as it contains only the interactions, genes with no interactions could potentiallt be missing).

I think the code:
`total_number_of_genes_polarity_module=len(np.unique(Interaction_Matrix.loc[:,'Interactor_Module']))`
would probably give the wrong number though (the output would be 2), so that would need to be modified.

Reply:
ok , so then it should be that, for example :  `total_number_of_genes_polarity_module=len(np.unique(Polarity_Related_Genes_Description))=123`  ?

Reply:
total_number_of_genes_polarity_module=len(np.unique(Polarity_Related_Genes_Description))=123